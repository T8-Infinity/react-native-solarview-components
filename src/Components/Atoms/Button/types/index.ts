import { TouchableOpacityProps } from 'react-native';

export interface IPropsButton extends TouchableOpacityProps {
  title: string | undefined;
  loading?: boolean;
  colorTitle?: string;
  onPress: (() => void) | undefined;
  backgroundColor?: string;
  outline?: boolean;
  disabled?: boolean;
  borderColor?: string;
  testID?: string;
}
