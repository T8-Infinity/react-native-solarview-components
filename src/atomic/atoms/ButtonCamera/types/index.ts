import { TouchableOpacityProps } from 'react-native';

export interface IButtonCameraProps extends TouchableOpacityProps {
  testID?: string;
  icon: string;
  onPress: () => void;
  color: string;
}
