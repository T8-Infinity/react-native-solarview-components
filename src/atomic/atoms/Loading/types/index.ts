import { ActivityIndicatorProps } from 'react-native';

export interface Props extends ActivityIndicatorProps {
  testID?: string;
  size: 'large' | 'small';
  color?: string;
}
