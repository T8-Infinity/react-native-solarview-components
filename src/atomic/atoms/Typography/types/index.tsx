import { fonts } from '@theme/fonts';
import { size } from '@theme/size';

export interface ITextProps {
  text: string;
  fontSize: keyof typeof size;
  color?: string;
  variant: keyof typeof fonts;
}
