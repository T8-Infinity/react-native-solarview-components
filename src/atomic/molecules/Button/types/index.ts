import { TouchableOpacityProps } from 'react-native';

export interface IPropsButton extends TouchableOpacityProps {
  title: string;
  loading?: boolean;
  colorTitle?: string;
  onPress: () => void;
  backgroundColor?: string;
  outline?: boolean;
  disabled?: boolean;
  borderColor?: string;
  testID?: string;
  icon: string;
}
