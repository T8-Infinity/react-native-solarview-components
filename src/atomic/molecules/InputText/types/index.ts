import { TextInput, TextInputProps } from 'react-native';

export interface InputTextProps extends TextInputProps {
  title: string;
  value: string;
  label?: string;
  icon?: string;
  secret?: boolean;
  touched?: boolean;
  errorMessage?: string;
  onChangeText: (e: any) => void;
  handlePressIcon?: () => void;
  colorTitle?: string;
}

export type Props = InputTextProps & {
  inputRef?: React.RefObject<TextInput>;
};

export interface InputTypesProps {
  colorTitle?: string;
  touched?: boolean;
}
