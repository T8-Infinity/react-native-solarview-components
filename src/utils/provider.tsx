import React from 'react';
import { ThemeProvider } from 'styled-components/native';
import { theme } from '@theme/index';

export const provider: React.FC = ({ children }: any) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};
