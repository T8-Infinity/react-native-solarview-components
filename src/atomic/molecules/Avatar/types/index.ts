import { IButtonCameraProps } from '@atomic/atoms/ButtonCamera/types';

export interface IAvatarProps {
  testID?: string;
  url: string;
  withLabel?: string;
  type: 'integrator' | 'owner';
  size: 'xs' | 'sm' | 'md' | 'lg' | 'xl';
  color: string;
  icon: string;
}

export type IProps = IAvatarProps & IButtonCameraProps;
