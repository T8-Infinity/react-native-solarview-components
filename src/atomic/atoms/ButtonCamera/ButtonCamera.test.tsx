import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import { provider as wrapper } from '@utils/provider';
import { IButtonCameraProps } from './types';

import { ButtonCamera } from '.';

describe('Atoms/ButtonCamera', () => {
  const elButtonCamera = 'ButtonCamera';

  const props: IButtonCameraProps = {
    color: 'red',
    icon: 'camera',
    onPress: jest.fn(),
  };

  const setup = () => {
    const utils = render(<ButtonCamera {...props} />, {
      wrapper,
    });
    const sut = utils.getByTestId(`${elButtonCamera}`);
    const sutButton = utils.getByTestId(`${elButtonCamera}_touch`);

    return {
      sut,
      sutButton,
      ...utils,
    };
  };

  it(`should render component ${elButtonCamera}`, () => {
    const { sut } = setup();
    expect(sut).toBeTruthy();
  });

  it(`should press button ${elButtonCamera}`, () => {
    const { sutButton } = setup();
    fireEvent(sutButton, 'press');
    expect(props.onPress).toHaveBeenCalledTimes(1);
  });

  it('should render snapshot', () => {
    const { toJSON } = setup();
    expect(toJSON()).toMatchSnapshot();
  });
});
