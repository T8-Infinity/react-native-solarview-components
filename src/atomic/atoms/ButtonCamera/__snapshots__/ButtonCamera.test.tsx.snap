// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/ButtonCamera should render snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "width": "100%",
      },
    ]
  }
  testID="ButtonCamera"
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="ButtonCamera_touch"
  >
    <View
      color="red"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "red",
            "borderRadius": 48,
            "height": 48,
            "justifyContent": "center",
            "width": 48,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 18,
            },
            Array [
              Object {},
            ],
            Object {
              "fontFamily": "material-community",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ó°„€
      </Text>
    </View>
  </View>
</View>
`;
